tests/codex/testerasure.nim

```diff
@@ -252,3 +252,22 @@ suite "Erasure encode/decode":
       decoded.treeCid == manifest.treeCid
       decoded.treeCid == verifiable.originalTreeCid
       decoded.blocksCount == verifiable.originalBlocksCount
+
+  for i in 1..5:
+    test "Should encode/decode using various parameters " & $i & "/5":
+      let
+        blockSize   = rng.sample(@[1, 2, 4, 8, 16, 32, 64].mapIt(it.KiBs))
+        datasetSize = rng.sample(@[1, 2].mapIt(it.MiBs))
+        ecK         = rng.sample(@[20].mapIt(it.Natural))
+        ecM         = rng.sample(@[10].mapIt(it.Natural))
```

Comment:
Unfortunately when debugging issue with threading I discovered another issue with erasure coding (even if 0 blocks are deleted from the store).
```
 Unhandled exception: Unable to encode manifest, not enough blocks, ecK = 20, blocksCount = 16 [InsufficientBlocksError]
```
When using ecK=20 and ecM=10 params, I haven't observed such issue, therefore these collections have only these single elements.

Reply:
This is not very surprising, given that we do the network-level EC on the block level. If there are less blocks than `ecK`, it breaks. Not sure what's the flow here, but can't you include this constraints in the above randomly generated test parameters?

Reply:
The problem is that in this scenario there are 0 blocks deleted, so it should always bee N blocks available (ecK + ecM) for decoding. Regardless of this, an error shown above occurs sometimes.

Reply:
It seems to me that the problem is with encoding, not decoding (that's what the quoted error message says?). Since there are less than K blocks, you cannot split it into K pieces. Though padding should solve that...

Reply:
You're right, I've been focused on decoding so much lately that I immediately assume it's about decoding. Indeed the error is trivial and expected.